nats-base-client/semver.ts

```diff
@@ -1,6 +1,6 @@
 export type SemVer = { major: number; minor: number; micro: number };
 export function parseSemVer(
-  s: string,
+  s = "",
 ): SemVer {
   const m = s.match(/(\d+).(\d+).(\d+)/);
```

Comment:
I suggest using https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string provided regex.

Reply:
I use this pervasively for some version-specific functionality - I rather keep it as I have it as it is hardened.